src/js/_enqueues/wp/updates.js

```diff
@@ -2536,6 +2570,32 @@
 			// Check the queue, now that the event handlers have been added.
 			wp.updates.queueChecker();
 		} );
+		
+		/**
+		 * Bulk action handler for posts/pages.
+		 *
+		 * Handles the scenario where no items are selected.
+		 *
+		 * @since 6.4.0
+		 *
+		 * @param {Event} event Event interface.
+		 */
+		$postFilterForm.on( 'click', '#doaction, #doaction2', function( event ) {
+			var itemsSelected = $postFilterForm.find( 'input[name="post[]"]:checked' );
+
+			// Bail if there were no items selected.
+			if ( ! itemsSelected.length ) {
+				event.preventDefault();
+
+				wp.a11y.speak( wp.i18n.__( 'Please select at least one item to perform this action on.' ) );
+
+				return wp.updates.addInlineNotice(
+					$(".bulkactions"),
+					false,
+					"Please select at least one item to perform this action on."
```

Comment:
I think it needs translation, or if possible removing duplication of this phrase to use it once